pkg/api/applications/v2/api_test.go

```diff
@@ -0,0 +1,246 @@
+
+					expAPI, err := experiments.NewAPIWithEndpoint(client, scn.Link(api.RelationExperiments))
+					require.NoError(t, err, "failed to create experiment API for application")
+
+					exp, err := expAPI.CreateExperimentByName(ctx, newExperimentName(), td.Experiment)
+					require.NoError(t, err, "failed to create experiment")
+					assert.NotEmpty(t, exp.Link(api.RelationTrials), "missing trials link")
+					assert.NotEmpty(t, exp.Link(api.RelationNextTrial), "missing next trial link")
+					assert.NotEmpty(t, exp.Link(api.RelationSelf), "missing self link")
+					assert.Equal(t, app.Name, exp.Labels["application"], "incorrect application label")
+					assert.Equal(t, scn.Name, exp.Labels["scenario"], "incorrect scenario label")
+
+					_, err = expAPI.CreateTrial(ctx, exp.Link(api.RelationTrials), experiments.TrialAssignments{
+						Labels:      map[string]string{"baseline": "true"},
+						Assignments: td.Baseline,
+					})
+					require.NoError(t, err, "failed to create baseline trial")
+
+					for {
+						ta, err := expAPI.NextTrial(ctx, exp.Link(api.RelationNextTrial))
+						var aerr *api.Error
+						if errors.As(err, &aerr) && aerr.Type == experiments.ErrExperimentStopped {
+							break
+						}
+						require.NoError(t, err, "failed to fetch trial assignments")
+						assert.NotEmpty(t, ta.Location(), "missing location")
+
+						err = expAPI.ReportTrial(ctx, ta.Location(), td.TrialResults(&ta))
+						require.NoError(t, err, "failed to report trial")
+					}
+				})
+			}
+
+			err = appAPI.DeleteActivity(ctx, ai.URL)
+			require.NoError(t, err, "failed to acknowledge activity")
+		}
+	})
+
+	t.Run("Request Activity", func(t *testing.T) {
+		if !ok {
+			t.Skip("skipping activity request.")
+		}
+
+		// TODO This should use CheckEndpoint to get the feed link via a HEAD request
+		lst, err := appAPI.ListApplications(ctx, applications.ApplicationListQuery{})
+		require.NoError(t, err, "failed to fetch the application list necessary for the feed URL")
+
+		// TODO Do we need to create this or is it implicit from create scenario?
+		sa := &applications.ScanActivity{
+			Scenario: scnMeta.Location(),
+		}
+		err = appAPI.CreateActivity(ctx, lst.Link(api.RelationAlternate), applications.Activity{Scan: sa})
+		require.NoError(t, err, "failed to request scan")
+
+		ra := &applications.RunActivity{
+			Scenario: scnMeta.Location(),
+		}
+		err = appAPI.CreateActivity(ctx, lst.Link(api.RelationAlternate), applications.Activity{Run: ra})
+		require.NoError(t, err, "failed to request run")
+		run.Add(1)
+	})
+
+	// Wait for all run activities to finish
+	run.Wait()
+	cancelSub()
+
+	t.Run("Delete Application", func(t *testing.T) {
+		if appMeta.Location() == "" {
+			t.Skip("skipping delete application.")
+		}
+
+		err := appAPI.DeleteApplication(ctx, appMeta.Location())
+		require.NoError(t, err, "failed to delete application")
+
+		// TODO Verify the experiment still exists via the /v1/experiments/ endpoint? Or will it be deleted?
```

Comment:
I was thinking these should still exist, but will be unreachable since the application is hidden. In my mind, the same thing applies to scenarios, templates (formerly scans) and activity items related to the application.

Reply:
Yeah, this ties into our conversation yesterday about unlabeled experiments: though this is slightly different in that we have "orphaned" experiments. It makes me wonder if the controller should be the one syncing the missing applications or not.